src/utils/message.ts

```diff
@@ -49,23 +60,57 @@ export const prepareErrorMessage = (id: number, clientId: string, error: unknown
   }
 };
 
-export const prepareMessage = (
-  id: string | number,
-  clientId: string,
-  data:
-    | ServerInfo
-    | AccountInfo
-    | string
-    | Responses['block_content']
-    | BalanceHistoryData[]
-    | TxTypes.TxIdsToTransactionsResponse[]
-    | TxTypes.TransformedTransaction
-    | UtxosWithBlockResponse[]
-    | { subscribed: boolean }
-    | { lovelacePerByte: number },
-): string => {
+export const prepareMessage = ({ id, clientId, data }: Responses): string => {
   const message = JSON.stringify({ id, type: 'message', data });
 
   logger.debug(`[${clientId}] Prepared response for msg id ${id} with length ${message.length}`);
   return message;
 };
+
+const checkType = (expected: ParamsTypes, value: unknown) => {
+  const valueType = typeof value;
+
+  switch (expected) {
+    case 'string | number': {
+      return valueType === 'string' || valueType === 'number';
+    }
+    case 'string[]': {
+      if (!Array.isArray(value)) return false;
+
+      for (const element of value) if (typeof element !== 'string') return false;
+
+      return true;
+    }
+    case 'details': {
+      return Details.includes(valueType as Details);
+    }
+  }
+
+  return valueType === expected;
+};
+
+export function assertParams(
```

Comment:
pls introduce unit tests for this function

Reply:
fcf302c should solve this